### ISSUE TITLE
SQLite's JSON operators don't retain boolean three-valued logic

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13220]
The SQLite boolean in JSON emulation does not maintain three-valued logic correctly. For example, this query is being generated in a test:

```sql
select (
  select coalesce(
    json_group_array(json_array(json(case
      when v0 then 'true'
      else 'false'
    end))),
    json_array()
  )
  from (
    select null as v0
    union all
    select 1 as a
  ) as t
)
```

When it should be this:

```sql
select (
  select coalesce(
    json_group_array(json_array(json(case
      when v0 then 'true'
      when not v0 then 'false' -- keep NULL values as NULL
    end))),
    json_array()
  )
  from (
    select null as v0
    union all
    select 1 as a
  ) as t
)
```

**Comment by lukaseder** [comment]
Or even better:

```sql
select (
  select coalesce(
    json_group_array(json_array(json(case v0
      when true then 'true'
      when false then 'false'
    end))),
    json_array()
  )
  from (
    select null as v0
    union all
    select 1 as a
  ) as t
)
```

**Comment by lukaseder** [comment]
JSON support has been added for the `SQLDialect.SQLITE` dialect with https://github.com/jOOQ/jOOQ/issues/11465 in jOOQ 3.16.0 only, so no backport to 3.15 will be needed.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0 and 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13221)
